src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -363,16 +363,33 @@ private void updateUninitializedVariables(Deque<DetailAST> prevScopeUnitializedV
     }
 
     /**
-     * If token is LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, or LITERAL_ELSE, then do not
-     * update the uninitialized variables.
+     * If token is LITERAL_IF and there is an {@code else} following or token is CASE_GROUP and
+     * there is another {@code case} following, then update the uninitialized variables.
      * @param ast token to be checked
      * @return true if should be updated, else false
      */
     private static boolean shouldUpdateUninitializedVariables(DetailAST ast) {
-        return ast.getType() != TokenTypes.LITERAL_TRY
-                && ast.getType() != TokenTypes.LITERAL_CATCH
-                && ast.getType() != TokenTypes.LITERAL_FINALLY
```

Comment:
@rnveach I add some more samples taken from diff report and write a specific input file for this fix. But to be honest, I am not sure whether it is enough for regression.

IMO, just a suggestion, we need think twice of the whole structure of this check. After investigated on it these days, I have found several part of codes which are much likely to have some unknown problems. The code in some early version (i.e. v6.13) is really clean, but maybe after many patches it is much more complicated now.

I think I could probably find more false positive bugs if I keep investigating. Maybe in the future we'd better to make a summary of all the conditions we need to concern about, and then refactor some part of this check. But it will need a lot of time and work obviously and not very easy. So maybe it is not necessary.